kotlinx-coroutines-core/common/src/channels/Produce.kt

```diff
@@ -141,3 +152,20 @@ internal open class ProducerCoroutine<E>(
         if (!processed && !handled) handleCoroutineException(context, cause)
     }
 }
+
+private class DropWhileBusyProducerCoroutine<E>(
+    parentContext: CoroutineContext,
+    channel: Channel<E>
+) : ProducerCoroutine<E>(parentContext, channel) {
+
+    private var hasSentFirstValue: Boolean = false
```

Comment:
Now, since this is a channel, the `send` function can be called concurrently, so this variable has to be atomic (`private val hasSentFirstValue = atomic(false)`) and the check & set on it has to be atomic (`hasSentFirstValue.getAndSet(true)`) to make sure that when there are two concurrent calls to `send` it works correctly.